compiler/qsc/src/bin/qsc.rs

```diff
@@ -69,12 +70,24 @@ fn main() -> miette::Result<ExitCode> {
 
     let entry = cli.entry.unwrap_or_default();
     let sources = SourceMap::new(sources, Some(entry.into()));
-    let (unit, errors) = compile(&store, &dependencies, sources, PackageType::Lib);
+    let package_type = if cli.emit.contains(&Emit::Qir) {
+        PackageType::Exe
```

Comment:
Not in this case, no. This will support emitting the generated QIR into a .ll file on disk, but doesn't specifically address any interop strategies for how to consume that code.  The assumption is the file would then be simulated with something like qir-runner (https://github.com/qir-alliance/qir-runner) or would be submitted to the Azure Quantum service for execution.